README.md

```diff
@@ -20,51 +20,84 @@ To ensure Circuit Maintenance is automatically re-installed during future upgrad
+Notification Sources are defined in two steps:
 
-<p align="center">
-<img src="https://raw.githubusercontent.com/nautobot/nautobot-plugin-circuit-maintenance/develop/docs/images/email_config.png" width="500">
-</p>
+#### 2.1 Define Notification Sources in `configuration.py`
 
-Attributes:
+In the `PLUGINS_CONFIG`, under the `nautobot_circuit_maintenance` key, we should define the Notification Sources that will be able later in the UI.
+
+There are two mandatory attributes (other keys are dependent on the integration type, and will be documented below):
+
+- `alias`: Name to identify the Source and will be available in the UI.
+- `url`: URL to reach the Notification Source (i.e. `imap://imap.gmail.com:993`)
+
+> Currently, only IMAP email box integration is supported as URL scheme,
+
+##### IMAP
+
+There are 2 extra attributes:
+
+- `account`: Identifier (i.e. email address) to use to authenticate.
+- `secret`: Password to IMAP authentication.
+
+> Gmail example: [How to setup Gmail with App Passwords](https://support.google.com/accounts/answer/185833)
+
+```py
+PLUGINS_CONFIG = {
+    "nautobot_circuit_maintenance": {
+        "notification_sources": [
+            {
+                "alias": "my custom alias",
+                "account": os.environ.get("CM_NS_1_ACCOUNT", ""),
+                "secret": os.environ.get("CM_NS_1_SECRET", ""),
+                "url": os.environ.get("CM_NS_1_URL", ""),
+            }
+        ]
+    }
+}
+````
+
+#### 2.2 Add `Providers` to the Notification Sources
 
-- Email: email address
-- Password: password to access the email service API for this service
-- URL: email service URL
-- Server Type: type of email service. Currently, only GMAIL is supported.
-- Providers: list of Ciruit Providers that will be tracked.
+In the Circuit Maintenance plugin UI section, there is a **Notification Sources** button (yellow) where you can configure the Notification Sources to track new circuit maintenance notifications from specific providers.
 
-> [How to setup Gmail with App Passwords](https://support.google.com/accounts/answer/185833)
+Because the Notification Sources are defined by the configuration, you can only view and edit `providers`, but not `add` or `delete` new Notification Sources via UI or API.
 
 ### 3. Run Handle Notifications Job
 
-There is an asynchronous task defined as a **Nautobot Job**, **Handle Circuit Mainentance Notifications** that will connect to the emails sources defined under the Settings section (step above), and will fetch new notifications received since the last notification was fetched.
+There is an asynchronous task defined as a **Nautobot Job**, **Handle Circuit Mainentance Notifications** that will connect to the emails sources defined under the Notification Sources section (step above), and will fetch new notifications received since the last notification was fetched.
```

Comment:
```suggestion
There is an asynchronous task defined as a **Nautobot Job**, **Handle Circuit Maintenance Notifications** that will connect to the emails sources defined under the Notification Sources section (step above), and will fetch new notifications received since the last notification was fetched.
```